server/src/settings/Settings.ts

```diff
@@ -0,0 +1,62 @@
+/** Describe the YML relevant settings part as defined in `package.json`. */
+export interface IYseopmlSettings {
+    yseopml: IYseopmlServerSettings;
+}
+
+/** Describe the _server_ relevant settings part as defined in `package.json`. */
+export interface IYseopmlServerSettings {
+    activateParsingProblemsReporting: boolean;
+    parseAllProjectFilesAtStartup: boolean;
+    pathToPredefinedObjectsXml: string;
+    ymlParsingIssueSeverityLevel: string;
+    kaoFiles: string[];
+    documentFormat: IDocumentFormatSettings;
+}
+
+/**
+ * Possible attributes that can be sent as document format settings as defined in `package.json`.
+ */
+export interface IDocumentFormatSettings {
+    // 'yes' and 'no' acts like a new type. Impossible to use const.
+    /** `yes` to enable the document format feature. `no` otherwise. */
+    enableDocumentFormat: 'yes' | 'no';
+    /** `true` to add semicolon when possible. `false` otherwise. */
+    semicolonWhenPossible: boolean;
+    /**
+     * `true` to set a one space distance between some keywords and parenthesis. \
+     * `false` to have this distance set to zero.
+     */
+    spaceBetweenKeywordAndParenthesis: boolean;
+    // More options to come.
+}
+
+/**
+ * Default values for the document format settings, as defined in `package.json`.
+ */
+export const DEFAULT_DOC_FORMAT_SETTINGS: IDocumentFormatSettings = {
+    enableDocumentFormat: 'yes',
+    semicolonWhenPossible: true,
+    spaceBetweenKeywordAndParenthesis: false,
+};
+
+/**
+ * Set in an instance of document format settings set the default values for every undefined options.
```

Comment:
Double “set”. + kinda the same confusion as in the AA IHM. (set a default value ≠ set a parameter to its default value)

```suggestion
 * In an instance of document format settings, set every undefined option to its default value.
```

Reply:
I prefer your suggestion. :+1: